Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -1,13 +1,20 @@
 import Foundation
+import Combine
 
 /// A global singleton which holds base configuration for the OpenFeature library.
 /// Configuration here will be shared across all ``Client``s.
 public class OpenFeatureAPI {
-    private var _provider: FeatureProvider?
```

Comment:
Why not initialise this to the NoOpProvider so we can remove the nullability?

Reply:
Looks like that's what they do in the [Java SDK](https://github.com/open-feature/java-sdk/blob/0f5ee6556ad1108da57a8969b46b191309ebb3b1/src/main/java/dev/openfeature/sdk/ProviderRepository.java#L24) (I personally use that one as reference quite often). Let's do it! Can we just tackle this in a subsequent PR?

Reply:
Yes, let's get this one merged and pickup the `NoOpProvider` approach in a separate one.